proto/redpanda/api/dataplane/v1alpha1/secret.proto

```diff
@@ -0,0 +1,230 @@
+syntax = "proto3";
+
+package redpanda.api.dataplane.v1alpha1;
+
+import "buf/validate/validate.proto";
+import "google/api/annotations.proto";
+import "google/api/field_behavior.proto";
+import "protoc-gen-openapiv2/options/annotations.proto";
+
+// Secret defienes the secret resource
+message Secret {
+  string id = 1 [
+    (google.api.field_behavior) = IMMUTABLE,
+    (google.api.field_behavior) = OUTPUT_ONLY
+  ];
+  map<string, string> labels = 2 [
+    (buf.validate.field).map = {
+      values: {
+        string: {pattern: "^([\\p{L}\\p{Z}\\p{N}_.:/=+\\-@]*)$"}
+      },
+    },
+    (google.api.field_behavior) = IMMUTABLE
+  ];
+}
+
+message SecretInput {
```

Comment:
created this due this limitation, https://grpc-ecosystem.github.io/grpc-gateway/docs/mapping/customizing_openapi_output/#:~:text=For%20IMMUTABLE%20and%20INPUT_ONLY%20fields%2C%20there%20is%20an%20open%20issue%20in%20the%20Open%20API%20specification%20for%20adding%20functionality%20for%20write%2Donce%20or%20immutable%20fields%20to%20the%20spec.